core/src/main/groovy/com/novoda/gradle/release/ReleasePlugin.groovy

```diff
@@ -49,26 +49,24 @@ class ReleasePlugin implements Plugin<Project> {
     void attachArtifacts(PublishExtension extension, Project project) {
         if (project.plugins.hasPlugin('com.android.library')) {
             project.android.libraryVariants.all { variant ->
-                def artifactId = extension.artifactId;
-                addArtifact(project, variant.name, artifactId, new AndroidArtifacts(variant))
+                addArtifact(project, variant.name as String, extension, new AndroidArtifacts(variant))
             }
         } else {
-            addArtifact(project, 'maven', project.publish.artifactId, new JavaArtifacts())
+            addArtifact(project, 'maven', extension, new JavaArtifacts())
         }
     }
 
 
-    void addArtifact(Project project, String name, String artifact, Artifacts artifacts) {
```

Comment:
Are you sure this is not breaking any public API?

Reply:
Which "api" do you mean?
this method was public before (however why) but it shouldn't be public.

No other plugin should use these method. Even if it has **this** plugin as dependency.
If another plugin wants to use our plugin (to create artifacts) it should just apply it and wrap their own logic over it.

So all in all:
This method was never designed to call from outside!

Reply:
My point is: we can't remove a public method all of sudden, even if there was no explicit intention to provide this as public API in the past. The right way of approaching this is to keep the method, annotate it as `@Deprecated` and remove it in a follow-up major version.